spark/src/main/kotlin/com/adevinta/spark/components/snackbars/Snackbar.kt

```diff
@@ -41,206 +36,281 @@ import androidx.compose.ui.Alignment
     onActionClick: (() -> Unit)? = null,
+    onDismissIconClick: (() -> Unit)? = null,
     content: @Composable () -> Unit,
 ) {
-    val contentColor = contentColorFor(backgroundColor = colors.baseColor)
-
-    val actionComposable: (@Composable () -> Unit)? = actionLabel?.let {
-        @Composable {
-            BaseSparkButton(
-                colors = ButtonDefaults.textButtonColors(contentColor = contentColor),
-                onClick = { onActionClick?.invoke() },
-                elevation = null,
-                content = { Text(it) },
-            )
-        }
-    }
-
-    val titleComposable: (@Composable () -> Unit)? = title?.let {
-        @Composable {
-            Text(
-                text = it,
-                maxLines = 2,
-            )
-        }
+    val backgroundColor = when (style) {
+        SnackbarStyle.Filled -> intent.filledColor
+        SnackbarStyle.Tinted -> intent.tintedColor
     }
+    val contentColor = contentColorFor(backgroundColor = backgroundColor)
 
     MaterialSnackBar(
         modifier = modifier
             .padding(16.dp)
             .sparkUsageOverlay(),
-        shape = if (LocalLegacyStyle.current) SparkTheme.shapes.extraSmall else SparkTheme.shapes.small,
-        actionOnNewLine = actionOnNewLine,
-        containerColor = colors.baseColor,
+        shape = SparkTheme.shapes.medium,
+        actionOnNewLine = isActionOnNewLine,
+        containerColor = backgroundColor,
         contentColor = contentColor,
-        action = actionComposable,
+        dismissAction = getDismissIconComposable(
+            intent = intent,
+            style = style,
+            onClick = { onDismissIconClick?.invoke() },
+            isDismissIconEnabled = isDismissIconEnabled,
+        ),
+        action = getActionComposable(
+            intent = intent,
+            onClick = { onActionClick?.invoke() },
+            actionLabel = actionLabel,
+            style = style,
+            isActionOnNewLine = isActionOnNewLine,
+        ),
     ) {
-        Column {
-            ProvideTextStyle(value = SparkTheme.typography.body2.highlight) {
-                titleComposable?.invoke()
-                Spacer(Modifier.height(4.dp))
+        Row(
+            verticalAlignment = Alignment.CenterVertically,
+            horizontalArrangement = Arrangement.spacedBy(4.dp),
+        ) {
+            icon?.let {
+                Icon(
+                    sparkIcon = it,
+                    contentDescription = null, // this is a decorative icon
+                    modifier = Modifier.size(24.dp),
+                )
             }
-            Row(verticalAlignment = Alignment.CenterVertically) {
-                if (title == null) {
-                    icon?.let {
-                        it(Modifier.size(24.dp))
-                        Spacer(Modifier.width(16.dp))
-                    }
+            content()
+        }
+    }
+}
+
+@Composable
+private fun getDismissIconComposable(
```

Comment:
Imo we should stay with the previous impl here since the button color will always be the content color whatever the style and the intent is.
Currently, as it is, it's difficult by reading the code to know in which case which button is used on by what requirement it can be shown or not